src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -176,23 +173,32 @@ void emit(ServiceEmitter emitter, Map<String, String[]> dimensions)
       }
     }
 
-    void emit(ServiceEmitter emitter, ServiceMetricEvent.Builder builder)
+    void emit(ServiceEmitter emitter, Map<String, String[]> dimensions)
     {
-      builder.setDimension("gcGen", name);
+      Map<String, String[]> dimensionsCopy = ImmutableMap
+          .<String, String[]>builder()
+          .putAll(dimensions)
+          .put("gcGen", new String[]{name})
+          .build();
 
       if (collector != null) {
-        collector.emit(emitter, builder);
+        collector.emit(emitter, dimensionsCopy);
```

Comment:
This and the following statements should be switched, I think. Also probably it makes sense to emit gcName not only for collector.emit(), but for spaces, i. e. it could be like
```java
ImmutableMap.Builder<String, String[]> ds = ImmutableMap
           .<String, String[]>builder()
           .putAll(dimensions)
           .put("gcGen", new String[]{name});
if (collector != null) {
  ds.put("gcName", ...);
}
dimensionsCopy = ds.build();
if (collector != null)
  collector.emit(emitter, dimensionsCopy);
```

Or in some more elegant way